src/measure_theory/measure_space.lean

```diff
@@ -2258,6 +2258,152 @@ lemma measurable.comp_ae_measurable [measurable_space δ] {f : α → δ} {g : 
 
 end
 
+section ae_seq
+/-!
+We define here tools to prove statements about limits (infi, supr...) of sequences of
+`ae_measurable` functions.
+Given a sequence of a.e.-measurable functions `f : ι → α → β` with hypothesis
+`hf : ∀ i, ae_measurable (f i) μ`, and a pointwise property `p : α → (ι → β) → Prop` such that we
+have `hp : ∀ᵐ x ∂μ, p x (λ n, f n x)`, we define a sequence of measurable functions `ae_seq hf p`
+and a set `ae_seq_set hf p`, such that
+* `μ (ae_seq_set hf p)ᶜ = 0`
+* `x ∈ ae_seq_set hf p → ∀ i : ι, ae_seq hf hp i x = f i x`
+* `x ∈ ae_seq_set hf p → p x (λ n, f n x)`
+-/
+variables {α β ι} [measurable_space α] [measurable_space β] {f : ι → α → β} {μ : measure α}
+  {p : α → (ι → β) → Prop}
+
+/-- If we have the additional hypothesis `∀ᵐ x ∂μ, p x (λ n, f n x)`, this is a measurable set
+whose complement has measure 0 such that for all `x ∈ ae_seq_set`, `f i x` is equal to
+`(hf i).some x` for all `i` and we have the pointwise property `p x (λ n, f n x)`. -/
```

Comment:
Instead of `(hf i).some`, could you use `(hf i).mk f`, which is designed for this and has a nice associated API, here and elsewhere in the PR?